apps/native/app/src/lib/passkeys/useAuthenticatePasskey.ts

```diff
@@ -0,0 +1,73 @@
+import { Passkey, PasskeyAuthenticationResult } from 'react-native-passkey'
+import { btoa } from 'react-native-quick-base64'
+import {
+  convertAuthenticationResultsToBase64Url,
+  convertBase64UrlToBase64String,
+  padChallenge,
+} from './helpers'
+import { useGetPasskeyAuthenticationOptionsLazyQuery } from '../../graphql/types/schema'
+import { preferencesStore } from '../../stores/preferences-store'
+
+const ONE_HOUR = 3600000
+
+export const useAuthenticatePasskey = () => {
+  const isSupported: boolean = Passkey.isSupported()
+  const { lastUsedPasskey } = preferencesStore.getState()
```

Comment:
~This will not automatically re-run when `lastUsedPasskey` changes. Should either use a hook, or just call `getState` inside `authenticatePasskey`.